File: apps/api/src/main/kotlin/com/nagi4757/ec/api/product/domain/model/Product.kt
package com.nagi4757.ec.api.product.domain.model

data class Product(
    val id: Long? = null,
    val name: String,
    val price: Long,
    val stockQuantity: Int,
    val imageUrl: String?,
    val description: String?
) {
    init {
        require(stockQuantity >= 0) { "stockQuantity must be greater than or equal to 0" }
    }
}

File: apps/admin-web/src/types/product.ts
export type Product = {
    id: number
    name: string
    price: number
    stockQuantity: number
    imageUrl?: string | null
    description?: string | null
}

export type CreateProduct = Omit<Product, 'id'>;
export type UpdateProduct = Partial<Omit<Product, 'id'>>;
